.github/workflows/dry_run_publish.yml

```diff
@@ -0,0 +1,43 @@
+name: Dry-run Publish
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+  # TODO: remove
```

Comment:
Why not addressing those left TODO comments in that PR?

Reply:
All TODOs marks which will be addressed (see the issue description, which has 2 more steps left unchecked)